doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,297 @@ Reasoning with inductive types
+   The :n:`@one_term` may contain holes that are denoted by “_”. In this case,
+   the tactic checks that all subterms matching the pattern in the conclusion
+   and the hypotheses are compatible and performs case analysis using this subterm.
 
-   .. tacv:: destruct @term in @goal_occurrences
+   .. todo PR Need more explanation about the tactic does when destruction_arg is
+      a one_term that's not just an identifier'
 
-      This syntax is used for selecting which occurrences of :token:`term`
-      the case analysis has to be done on. The :n:`in @goal_occurrences`
-      clause is an occurrence clause whose syntax and behavior is described
-      in :ref:`occurrences sets <occurrencessets>`.
+   :n:`{+, @induction_clause }`
+     Giving multiple :n:`@induction_clause`\s is equivalent to applying :n:`destruct`
+     serially on each :n:`@induction_clause`.
 
-   .. tacv:: destruct @term {? with @bindings } {? as @or_and_intropattern_loc } {? eqn:@naming_intropattern } {? using @term {? with @bindings } } {? in @goal_occurrences }
-             edestruct @term {? with @bindings } {? as @or_and_intropattern_loc } {? eqn:@naming_intropattern } {? using @term {? with @bindings } } {? in @goal_occurrences }
+   :n:`@natural`
+     Equivalent to :tacn:`intros` :n:`until @natural` followed by `destruct` applied to
+     the last introduced hypothesis.
 
-      These are the general forms of :tacn:`destruct` and :tacn:`edestruct`.
-      They combine the effects of the ``with``, ``as``, ``eqn:``, ``using``,
-      and ``in`` clauses.
+     .. note::
+        To destruct a number, use the syntax :n:`destruct (@natural)` (not
+        very interesting anyway).
 
-.. tacn:: case @term
-   :name: case
+   :n:`@ident` (in :n:`@destruction_arg`)
+     If :token:`ident` is a quantified variable in the conclusion
+     of the goal, then :n:`destruct @ident` behaves like :tacn:`intros`
+     :n:`until @ident; destruct @ident`.
+     If :token:`ident` is a hypothesis of the context and :token:`ident`
+     is no longer dependent in the goal after application
+     of :tacn:`destruct`, it is erased (to avoid erasure, use parentheses, as
+     in :n:`destruct (@ident)`).
+
+   :n:`using @one_term {? with @bindings }`
+     Makes the tactic equivalent to :tacn:`induction` :n:`@induction_clause_list`.
+
+   :n:`@as_or_and_ipat`
+      Provides names for the variables introduced
+      into the context.  The :token:`as_or_and_ipat` must have the
+      form :n:`[ {*| {* @intropattern } } ]` with one :n:`{* @intropattern }`
+      for each constructor in the order in which the constructors are defined.
+      The items in
+      each :n:`{* @intropattern }` are used to name the variables created in
+      each new goal (one per constructor).  If there are not enough names,
+      Coq picks fresh names.
+
+   :n:`eqn : @naming_intropattern`
+      Defines new items in the context for the matching terms using the specified
+      or generated names.
```

Comment:
This is very unclear. What it should say is that it generates a new hypothesis which will be an equality between the term being case-analyzed and the constructor (applied to its arguments) corresponding to the current case. See my example above.

Reply:
:n:`eqn : @naming_intropattern`
      Generates a new hypothesis in each new subgoal that is an equality between
      the term being case-analyzed and the associated constructor (applied to
      its arguments).  The name of the new item may be specified in the
      :n:`@naming_intropattern`.